src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupFeature.java

```diff
@@ -0,0 +1,289 @@
+package org.broadinstitute.hellbender.utils.codecs.sampileup;
+
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.util.StringUtil;
+import htsjdk.tribble.Feature;
+
+import java.util.List;
+
+/**
+ * A tribble feature representing a SAM pileup.
+ *
+ * Allows intake of both simple (6-column) or extended/consensus (10/13-column) pileups. Simple pileup features will
+ * contain only basic information, no observed alleles or variant/genotype inferences, and so shouldn't be used as input
+ * for analysis that requires that information.
+ *
+ * @author mhanna
+ * @version 0.1
+ */
+public class SAMPileupFeature implements Feature {
+    public enum VariantType { NONE, SNP, INSERTION, DELETION, INDEL };
```

Comment:
If it's ok to drop the deprecated pileup formats than much of this class can also be dropped.